### ISSUE TITLE
studio: assets (are already) private

### ISSUE THREAD

**Issue by peterpeterparker** [deckgo/deckdeckgo#585]
### Affected part of DeckDeckGo
- [X ] Studio
 
### Feature Description

Users are editing their presentations privately but currently, when they upload assets (images or charts data), these are made public because the Firebase Storage rules makes the storage content public to everyone (only read).

Users are informed about this through a popup (see screenshot).

That's the current status but it's about to change :)

Goal of this feature is to make the presentations assets (images and data), which would be saved in the Firebase Storage, private too until the presentation is shared online ("published") respectively until the presentation becomes public.

### Screenshot

<img width="872" alt="Capture d’écran 2020-01-13 à 20 41 24" src="https://user-images.githubusercontent.com/16886711/72286802-2d123500-3646-11ea-93d9-40e0231706a1.png">

### Limitation

Google Firebase Storage is a bit tricky when it comes to having files private and on demand becoming public. One approach was described on [Stackoverflow](https://stackoverflow.com/questions/59006071/firebase-storage-security-rules-to-allow-public-or-owner-private-file-access) but will have, depending of the amount of assets, some impact on the performances.

Therefore we will also add a delay between the publication and listing of the presentation on the "feed" mask.

# UPDATE AND FINAL RESULTS

Actually assets are already *private*. This [stackoverflow](https://stackoverflow.com/a/59782959/5404186) answer the question.

Basically: yes all Google Storage files are exposed (read) on internet through a download URL but it contains a `token` which is almost impossible to guess, therefore, as long as rules are correctly set to read and write the data directly (without download url), then files could be considered as private.
